include/openssl/opensslconf.h.in

```diff
@@ -65,12 +65,12 @@ extern "C" {
  */
 # ifndef DECLARE_DEPRECATED
 #  define DECLARE_DEPRECATED(f)   f;
-#  ifdef __GNUC__
-#   if __GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ > 0)
+#   if (defined(__GNUC__) && (__GNUC__ > 3 || \
```

Comment:
I seem to recall there being an issue with checking if a macro is defined and then using it all in the same expression...or did I imagine that?? Anyone?

Reply:
I was wondering about that...  however, looking around in our code base, we're not exactly consistent.  I think it was something triggered by only some compilers, and I've no idea which it was.

But either way, yeah, we should continue to do that, thanks for spotting it.  I'll withdraw my approval.

Reply:
okay, changed the code to comply to the OpenSSL style. Is there a style guide somewhere BTW ?

Reply:
Yes there is - although it is in serious need of some update because there are all sorts of things (like this one) that don't appear in it:
https://www.openssl.org/policies/codingstyle.html